.github/workflows/test.yml

```diff
@@ -1,5 +1,9 @@
 name: Tests
-on: [pull_request, push]
+on:
+  pull_request:
+    paths:
+      - '**.py'
+      - 'poetry.lock'
```

Comment:
This should be all we need for now and will help reduce the number of builds. It seems to have worked for this PR.